### ISSUE TITLE
Windows jobs raising low memory warnings

### ISSUE THREAD

**Issue by jakirkham** [conda-forge/numpy-feedstock#334]
We are seeing several Windows jobs here struggling with memory usage. Here is a screenshot of several

![Image](https://github.com/user-attachments/assets/d50c9d1f-e73e-4206-9cc9-8fb1df0618da)

Looking at [one log]( https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1052179&view=logs&j=5be07ae1-d8ba-5406-47b6-8e3a3a12f825&t=c94d5231-acfc-5e67-dafc-77bdd383ead4&l=4879 ), we see warnings like this:

```
##[warning]Free memory is lower than 5%; Currently used: 96.87%
##[warning]Free memory is lower than 5%; Currently used: 96.85%
##[warning]Free memory is lower than 5%; Currently used: 95.34%
##[warning]Free memory is lower than 5%; Currently used: 96.85%
##[warning]Free memory is lower than 5%; Currently used: 96.85%
```

**Comment by rgommers** [comment]
Seems like someone introduced a bad/expensive test in the test suite.

The runners do seem to be low on memory; the end of the log says:
```
Maximum memory usage observed: 2.8G
```
which isn't totally unreasonable.

**Comment by jakirkham** [comment]
Thanks Ralf! 🙏

It could be

Have seen similar warnings on other feedstock builds recently (admittedly not all feedstock builds have this issue). So am wondering if there are also changes on the CI side (like image size), which have pushed some feedstocks closer to the memory limit

**Comment by h-vetinari** [comment]
I don't think it's numpy specific. I see these warnings across many feedstocks. It's likely to me that Azure introduced more visible/noisy warnings about this, without any real underlying change. Alternatively, perhaps they down-sized the agents somehow. But as long as jobs are not failing due to getting OOM-killed, I don't think there's something to do right now.

**Comment by jakirkham** [comment]
Yes am thinking Azure made some kind of change to the workers. See NumPy more as [a canary in the coal mine]( https://www.smithsonianmag.com/smart-news/what-happened-canary-coal-mine-story-how-real-life-animal-helper-became-just-metaphor-180961570/ )

Though it is quite high. The warnings above are approaching 97% memory usage. Have in some cases needed to restart because of this

Think we will need to do something about this. Though it probably will not be NumPy specific (unless I've missed some test detail that Ralf alluded to above)
